src/field_theory/minimal_polynomial.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Johan Commelin
+-/
+
+import ring_theory.integral_closure
+
+universes u v w
+open polynomial set function
+
+noncomputable theory
+
+/- Turn down the instance priority for subtype.decidable_eq and use classical.dec_eq everywhere,
+  to avoid diamonds -/
+local attribute [instance, priority 0] subtype.decidable_eq
+
+variables {α : Type u} {β : Type v}
+
+section min_poly_def
+variables [decidable_eq α] [decidable_eq β] [comm_ring α] [comm_ring β] [algebra α β]
+
+def minimal_polynomial {x : β} (hx : is_integral α x) : polynomial α :=
+well_founded.min polynomial.degree_lt_wf _ (ne_empty_iff_exists_mem.mpr hx)
+
+end min_poly_def
+
+namespace minimal_polynomial
+
+section ring
+variables [decidable_eq α] [decidable_eq β] [comm_ring α] [comm_ring β] [algebra α β]
+variables {x : β} (hx : is_integral α x)
+
+protected lemma monic : monic (minimal_polynomial hx) :=
```

Comment:
Why is everything `protected`? Perhaps I'm just confused about what `protected` does, but it seems the only effect here is to stymie someone who wants, for some reason, to use `open minimal_polynomial`.

Reply:
Fixed